### ISSUE TITLE
Dynamic osgb models?

### ISSUE THREAD

**Issue by mschneider1711** [esmini/esmini#277]
Is there a way to dynamically adjust the sizes of the osgb models to fit a given bounding box?

**Comment by eknabevcc** [comment]
Yes, in the ScenarioObject definition in the xosc file you can control the scaling by setting property "scaleMode" to one of three values: None, BBToModel, ModelToBB 
None: Bounding box and model will not be modified
BBToModel: Scale bounding box to 3D model
ModelToBB: Scale 3D model to specified bounding box

Example:
The following setting for pedestrian in [resources/xosc/pedestrian.xosc](https://github.com/esmini/esmini/blob/master/resources/xosc/pedestrian.xosc) 
```
  <Pedestrian mass="80" model="EPTa" name="pedestrian_adult" pedestrianCategory="pedestrian" model3d="../models/walkman.osgb">
     <ParameterDeclarations/>
     <BoundingBox>
        <Center x="0.06" y="0.0" z="1.9"/>
        <Dimensions height="3.8" length="1.5" width="0.4"/>
     </BoundingBox>
     <Properties>
        <!-- Scale modes: None, BBToModel, ModelToBB -->
        <Property name="scaleMode" value="ModelToBB"/>  
     </Properties>
  </Pedestrian>
```
will look like:
![image](https://user-images.githubusercontent.com/29146839/165600099-ed11f20f-5d63-4416-8c87-8f16b1a0f2f7.png)

More examples in [resources/xosc/Catalogs/Vehicles/VehicleCatalog.xosc](https://github.com/esmini/esmini/blob/master/resources/xosc/Catalogs/Vehicles/VehicleCatalog.xosc)

It only semi-dynamic. I.e. it's only adjusted at load time. If bounding box is modified later during simulation no adjustment of model will happen. Or the other way around, OpenSCENARIO bounding box will not be updated if the 3D model is scaled during the simulation.
